### DISCUSSION TITLE
What are the best models for clean vocal extract?

### DISCUSSION THREAD

**Discussion by momoxsoso** [Anjok07/ultimatevocalremovergui#689] [Q&A]
The models is the part that confuse me and i need help with the best ones for vocal extract. pls help.

**Comment by elleEnjolras** [comment]
waiting for answers as well

**Reply by schubachr-design** [reply]
how can i get the model that removes crowd noise please?

**Comment by LucidFir** [comment]
I just had a good success going in this order.

Take your song (vocals, instrumentals, all of it, the original).

Run it through UVR5 with the (MDX) Kim Vocals 2 model.

Take the vocal output and run THAT through the (VR) de-reverb model.

Take the output from that and run it through the (VR) aggressive de-echo model.

Take the output from that and run it through a (VR) karaoke model.

**Reply by soucarlosxavier** [reply]
I've had a great result running the audio thru Kim Vocals 2 them taking the exported audi and running againg using HP-Karaoke-UVR. It cleaned the minor artifcats the Kim Vocals 2 model left.

**Reply by ElvarP** [reply]
This worked really well for me. Thanks.

**Reply by celeste-ex3** [reply]
One H*ll of a combo Cleanest Vocals That I Have Made So Far, bro can you please state a formula for the instrumentals too like only instruments no vocals it would be highly appreciated. and this sh*t is just awsome. thanks again

**Reply by can-ylmaz** [reply]
Kinda late on the party, but worked pretty well for me as well, tysm!

**Reply by Skulp55** [reply]
can't i just use ensemble mode and select all of them

**Reply by LucidFir** [reply]
Why don't you try it and find out :)

**Reply by GodAtum360** [reply]
Hi, are you talking about using UVR-DeEcho-DeReverb, then UVR-De-Echo-Aggressive?

**Reply by wooley553** [reply]
I have found that just using MDX Kim Vocals 1 or 2 and then VR UVR-De-Echo-Normal does a better job with fewer processing than the 4 steps posted earlier. I have been laid up in bed for a week and have experimented extensively, but it seems there are many different approaches to accomplish good results but the two processing methods I have posted do it much quicker and simpler.

**Reply by wooley553** [reply]
Under CHOOSE VR MODEL, just download the "UVR-De-Echo-Normal" from the "VR Arch" Download Center

**Reply by DCWelch** [reply]
Thanks @wooley553 for the advice from your experimentation! Are you referring to the model by FoxJoy (below) for your second stage?
![image](https://github.com/user-attachments/assets/605bfffc-660d-480f-b9e5-443ae372082e)

**Comment by Aviteshmurmu19** [comment]
Thanks [LucidFir](https://github.com/LucidFir) and [soucarlosxavier](https://github.com/soucarlosxavier)
Just adding up what they said!
![UVR](https://github.com/Anjok07/ultimatevocalremovergui/assets/133533261/21641e41-9e4d-405d-a909-87b2715f597a)

**Reply by dgoryeo** [reply]
@Aviteshmurmu19 , @soucarlosxavier thanks for the very helpful comments and diagram. Am I right to think that for clean vocals from a movie, it would be sufficient to do just the first and last steps? Any thoughts and pointers would be helpful --I'm not an expert in audio engineering. Thanks.

**Reply by ghost** [reply]
@dgoryeo  That depends on what you are trying to achieve. But what ever that is, if at all possible, try to work ONLY on the center channel if you have a surround soundtrack available, that should give the cleanest results. If you try to extract movie vocals for RVC training, you should still use the de-echoing step. (I prefer UVR-DeEcho-DeReverb in the VR model section, but what model gives you the cleanest results depends heavily on the input audio).

@Aviteshmurmu19 ~~What is~~ [MDX-NET Reverb467](https://pixeldrain.com/u/UWn7d2iH) ? 
~~Can't find that in the download center~~

**Reply by Moshist2025** [reply]
Is this still the best method or are some of the later versions of Karaoke etc. better?

**Reply by akilmaf** [reply]
Thanks for the amazing tips! Still a noob question: I just want to get the voice of a narrator. Is there a simple 1 model that I can use which will give a solid voice? Call it a model for noobs. I got the order above though :D

**Reply by unityconstruct** [reply]
> Thanks for the amazing tips! Still a noob question: I just want to get the voice of a narrator. Is there a simple 1 model that I can use which will give a solid voice? Call it a model for noobs. I got the order above though :D

n00b here too, however: Theoretically, [less processing] = [more of what you don't want]
- each step `sculpts` the vocals away from the main-source
- while 1st/last might get to the point quicker, they will carve out less (from vocals in this diagram)

RE: Diagram
- This is a use case for cleaning vocals
- So if Instrumental is wanted, then work down that chain - realizing that different model tools will be used to carve the vocals out

As an audio dude:
- Seems like each iteration of processing is going to bastardize the source so seems like less processing is better
- But there's software out there for mastering that use ai algos, so suppose 'they've figured this out' and further processing might actually be improving things along the way.
- video upscaling bastardizes the main-source, but clearly for the better.
- Thus, the stereotypical `analog snobbery` might actually be exactly that.

Anyway, THANKS @Aviteshmurmu19  for the diagram, it gave me the clarity on what these models do & how to use them properly.

**Reply by sarahmanning190** [reply]
Is this still relevant on 2026? Are these the best models to this date?

**Comment by Jay987654321** [comment]
@soucarlosxavier hello sir saw you comment. "I've had a great result running the audio thru Kim Vocals 2 them taking the exported audi and running againg using HP-Karaoke-UVR. It cleaned the minor artifacts the Kim Vocals 2 model left."     How did you just select one. When I try to do it says select two model. Please can you explain the process

**Reply by celeste-ex3** [reply]
it;s because you are using ensabled mode switch to other modes which has the required model you wish to run and then try it

**Comment by Jay987654321** [comment]
<img width="678" alt="Screenshot 2024-04-07 at 10 18 03 AM" src="https://github.com/Anjok07/ultimatevocalremovergui/assets/65005060/64be9a3b-d620-40fb-85df-ee91dbd7779f">
Am I selecting the right setting for vocal and ins extracting? Please guide if I need to do some additional settings to improve the vocal and instrumental quality. Thanks in advance

**Comment by celeste-ex3** [comment]
well you have to do it one by one like furst run the whole thing through one model and then the result through the other model model and then the output of that model through the other one step wise as shown in this [comment](https://github.com/Anjok07/ultimatevocalremovergui/discussions/689#discussioncomment-6644349) that's how it is done, and change the process mode to mdx for kim vocals from ensambled mode, and vr for deverb that how you only select one model from each mode. if you didn't understod just ask and i'll try my best next time to provide ya pics along with easy to follow guide

**Reply by Jay987654321** [reply]
I got it man thank you so very much. Can you also please tell which is the bestest model to separate just the beats (crystal clear) from any track whether instrumental or with vocals

**Reply by Jaimison-Git** [reply]
What Segment Size and overlap?

**Comment by vikkymusik** [comment]
![Screenshot 2024-09-14 234104](https://github.com/user-attachments/assets/1102f174-a122-49ee-8b23-70c91b38f797)
guys,,, why i have only this less options in "available models"? please help me

**Reply by d3m0n1alamercernaria** [reply]
hey babe press the wrench next to start processing then download center, then vr, and download any one of those models you see

**Reply by Hlebuw3k** [reply]
Choose "Multi-stem Ensemble" in the main stem pair dropdown instead of "Vocals/Instrumental"

**Comment by wooley553** [comment]
Besides removing vocals are there any plugins for uvr5 to create a center channel and LFE from a stereo track for use in making a 5.1 surround mix?

**Comment by michaelcasado** [comment]
I've done the steps in the guide, but DeEcho always result in metallic blip-shatter sounds in the final version. Tested normal, aggressive, and the DeEcho-DeReverb models. Anyone else having similar problems or know what this could be?

**Reply by DCWelch** [reply]
I don't know exactly what you mean by "metallic blip-shatter sounds", but this isn't an exact science and sometimes the models struggle distinguishing between certain sounds in specific circumstances. You can play around to hopefully find something better suited for your application, and I hope you do! Most of the suggestions in this thread are great starting points that work really well in a majority of circumstances, but aren't perfect for all cases.

Personally, I think [wooley553](https://github.com/wooley553)'s suggestion is the best general blend of simple and effective.

Most of the time for me I use Kim Vocal 2 as a starting point, and oftentimes just that is perfectly acceptable... If not, then depending on the results and what I'm looking for, playing with some of the variations in [Aviteshmurmu19](https://github.com/Aviteshmurmu19)'s summary can preserve/remove the remaining qualities you want from there.

Good luck!

**Reply by michaelcasado** [reply]
I've used UVR for a while and manage to get good results. But its on the models to remove delay/echo as some call it. These do not work when I try them. These noise best described as the glitch noise that you could get when downloading Mp3 files back in the days ;) This noise is all over the file, even over the vocals. The models remove the Delay FX on the vocals, that is working - but this noise is all over both channels. Listen here on a short sample.  https://drive.google.com/file/d/1NCMjftFh4-wByfuanenDBXn3B2PCgyPg/view?usp=drive_link

**Reply by DCWelch** [reply]
Ahh, I see... Sorry for the misinterpretation.

If it means anything, I get reasonable results with most DeEcho / Normalize / DeReverb models that I've tried through UVR without this weird noise...

I can't listen to your sample as I don't have access, but if it's truly all over and obnoxious, then something (I have no idea what) might be up with your setup. I'd suggest trying to start fresh somehow, or if you share more specifics, maybe I can try to replicate.

Or maybe someone else knows...

**Reply by michaelcasado** [reply]
Sorry, I missed to open access to link. It’s fixed now so anyone can see what I mean.

for some reason when using essemble mode I don’t get these crazy stuff. But this mode isn’t optimal with my understanding since I can not control the order of the chain. But Kim2, karaoke, and then deEcho-DeReverb output a good copy of the vocal in an okey dry version.

I wanted to get as dry as possible and without harmonies, adlibs etc so I can use these to train my own vocal model.

I stumbled on the karaoke models since it can be possible to separate backing vocals - but what it also did was separate duets / rapper and singer that might overlap; happy surprise!

**Reply by apple73388-debug** [reply]
@michaelcasado
Hello brother, how are you? May I ask you about something?

**Reply by michaelcasado** [reply]
Just ask

**Comment by AZbitrate** [comment]
What about extracting Friday Night Funkin' vocals from YouTube videos where the song is posted in full?

**Comment by audio-guy** [comment]
Which model do you use on a single vocal mic to reduce bleed?
I’ve found that MDX Inst HQ5 often causes “chirping” artifacts, while the Model Denoise tends to choke certain parts of the vocal.

**Comment by romanr** [comment]
what is good in 2026?

**Reply by sarahmanning190** [reply]
x2

**Comment by wooley553** [comment]
If it's IOS, it's just plain stupid...follow the Apple sheep assholes
 
 
  On Mon, Jun 8, 2026 at 10:58 PM, ***@***.***> wrote:   


Is this still relevant on 2026? Are these the best models to this date?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications, keep track of coding agent tasks and review pull requests on the go with GitHub Mobile for iOS and Android. Download it today!
